backend/internal/service/jwt_service.go

```diff
@@ -6,92 +6,164 @@ import (
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/base64"
-	"encoding/pem"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"log"
-	"math/big"
 	"os"
 	"path/filepath"
 	"slices"
 	"strconv"
 	"time"
 
 	"github.com/golang-jwt/jwt/v5"
+	"github.com/lestrrat-go/jwx/v3/jwa"
+	"github.com/lestrrat-go/jwx/v3/jwk"
+
 	"github.com/pocket-id/pocket-id/backend/internal/common"
 	"github.com/pocket-id/pocket-id/backend/internal/model"
+	"github.com/pocket-id/pocket-id/backend/internal/utils"
 )
 
 const (
-	privateKeyFile = "jwt_private_key.pem"
+	privateKeyFilePem = "jwt_private_key.pem"
+	privateKeyFileJwk = "jwt_private_key.json"
+
+	// Size, in bits, of the RSA key to generate if none is found
+	rsaKeySize = 2048
```

Comment:
Kept it at 2048 for now. It can be increased to 3072 in a subsequent PR if you wish.